gui/src/FourPanelView/Scene3DPanelView/Scene3DPanelView.tsx

```diff
@@ -185,19 +182,35 @@ const Scene3DPanelView: FunctionComponent<Props> = ({grid, focusPosition, surfac
         }
     }, [grid, surfacesData])
 
-    const {camera, controls} = useMemo(() => {
-        if (!container) return {camera: undefined, controls: undefined}
-        const p = {x: (bbox.min.x + bbox.max.x) / 2, y: (bbox.min.y + bbox.max.y) / 2, z: (bbox.min.z + bbox.max.z) / 2}
+    const camera = useMemo(() => {
+        if (!container) return undefined
         const camera = new THREE.PerspectiveCamera( 45, width / height, 1, 100000 )
-        camera.position.set(p.x, p.y, p.z + (bbox.max.z - bbox.min.z) * 3)
+        const pointLight = new THREE.PointLight(0xffffff, 0.7)
+        camera.add(pointLight)
+        return camera
+    }, [container, width, height])
+
+    const controls = useMemo(() => {
+        if (!container) return undefined
+        if (!camera) return undefined
         const controls = new OrbitControls( camera, container )
-        controls.target.set(p.x, p.y, p.z)
-        addThreePointLights(camera, false, bbox.max.x - bbox.min.x, bbox.max.y - bbox.min.y, bbox.max.z - bbox.min.z)
-        return {
-            camera,
-            controls
-        }
-    }, [width, height, bbox, container])
+        return controls
+    }, [camera, container])
+
+    const center = useMemo(() => {
+        return {x: (bbox.min.x + bbox.max.x) / 2, y: (bbox.min.y + bbox.max.y) / 2, z: (bbox.min.z + bbox.max.z) / 2}
+    }, [bbox])
+
+    const zSpan = useMemo(() => {
+        return bbox.max.z - bbox.min.z
+    }, [bbox])
+
+    useEffect(() => {
+        if (!camera) return
+        if (!controls) return
+        camera.position.set(center.x, center.y, center.z + zSpan * 3)
+        controls.target.set(center.x, center.y, center.z)
+    }, [center.x, center.y, center.z, zSpan, camera, controls]) // important to not pass in center, because reference will change whenever bbox reference changes
```

Comment:
Regarding this comment, I agree that avoiding the reference to `center` is the right thing to do, but this got me looking at `bbox` to see why its reference would change so often (since it's memoized). I see that it's defined above as depending on the `grid` and the `surfaceData`. The former should be pretty stable; the latter is more volatile because the `scalarField` and `vectorField` parameters refer to whatever field is currently selected for display--thus might change a bit. (Obviously we want to keep the behavior where we recompute the bounding box if we stop showing some of the surfaces.)

Just wondering if it would be possible to render the `bbox` value more stable, maybe through memoizing the set of `surface` values or something? It's likely not worth the trouble when the `bbox` computation isn't very expensive and only happens on interaction with the controls (not the camera), but I thought I'd mention the possibility in case it seems useful in the future.